plugins/dicom_metadata_extractor/server/dicom_metadata_extractor.py

```diff
@@ -0,0 +1,129 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+
+###############################################################################
+#  Copyright Kitware Inc.
+#
+#  Licensed under the Apache License, Version 2.0 ( the "License" );
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#    http://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+###############################################################################
+
+import six
+import dicom
+
+try:
+    from girder.utility.model_importer import ModelImporter
+
+except ImportError:
+    ModelImporter = None
+
+
+class DicomMetadataExtractor(object):
+    def __init__(self, path, itemId):
+        """
+        Initialize the metadata extractor.
+
+        :param path: path of file from which to extract metadata on client or
+        server
+        :param itemId: item ID of item containing file on server
+        """
+        self.itemId = itemId
+        self.path = path
+        self.metadata = None
+
+    def extractMetadata(self):
+        """
+        Extract metadata from file on client or server and attach to item on
+        server.
+        """
+        self._extractMetadata()
+
+        if self.metadata is not None:
+            self._setMetadata()
+
+    def _extractMetadata(self):
+        """
+        Extract metadata from file on client or server using hachoir-metadata
+        or pydicom for DICOM images.
+        """
+
+        try:
+            dcMetadata = dicom.read_file(self.path, stop_before_pixels=True)
```

Comment:
Does this plugin only work with filesystem assetstore? It'd be nice if it worked with any assetstore.